lib/serverengine/socket_manager.rb

```diff
@@ -0,0 +1,125 @@
+#
+# ServerEngine
+#
+# Copyright (C) 2012-2013 Sadayuki Furuhashi
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+require 'socket'
+require 'fcntl'
+
+module ServerEngine
+  module SocketManager
+
+    def self.new_socket_manager
```

Comment:
I fixed it now. 
It looks easy but it was a little confusing.

Firstly I thought I I have to use OptionParser not using ARGV[-1].
But if I use OptionParser here, in fluentd side it is also used, I have to set all option in both side not to cause OptionParser::InvalidOption. it is bored.
Then, secondly I thought I have to create interface class of command parser and use this interface in fluentd side. like this 
```ruby
opt_parser = ServerEngine::OptParser.new
```
But I did't want users to oblige to use bored setting like above because it is too bored and not cool.

Then at last, I used ENV unwillingly.
If I should do in other way, I'll do.